### PR TITLE
Add preliminary wrapper for Python class

### LINKED ISSUE #1493: Add import of a class type in the C-Python wrapper
### Describe the feature
The most basic support should be added for classes allowing just the importation of the class type object.

### Test Code
Provide code which does not currently work but which should do when this issue is fixed:
```python
from module import class
```

### Proposed Solution
A `PyClassDef` class should be created in `pyccel/ast/cwrapper.py`. This object will be built in the class `CToPythonWrapper` in a new function `_visit_ClassDef`.

In the `CWrapperPrinter` the function `self._print_PyClassDef` should be added printing the following code from the information in the `PyClassDef` instance:
```c
typedef struct {
    PyObject_HEAD
    <class_name>* instance;
} Py<class_name>Object;

static PyTypeObject Py<class_name>Type = {
    PyVarObject_HEAD_INIT(NULL, 0)
    .tp_name = "<module_name>.<class_name>",
    .tp_doc = PyDoc_STR("<class_docstring>"),
    .tp_basicsize = sizeof(Py<class_name>Object),
    .tp_itemsize = 0,
    .tp_flags = Py_TPFLAGS_DEFAULT,
    .tp_new = PyType_GenericNew,
};
```

The function `CToPythonWrapper._build_module_exec_function` should be modified to add the new object `PyTypeObject Py<class_name>Type` to the module.

### Additional Context

This step is equivalent to the first step of this tutorial : https://docs.python.org/3/extending/newtypes_tutorial.html#the-basics

### PR DESCRIPTION
**Pull request by EmilyBourne**
Add preliminary support for importing classes. Classes are added to modules but no class methods or variables are added in this PR. This allows us to import the class but not to interact with it. Fixes #1493

### CODE REVIEW

**Comment by EmilyBourne** [comment]
/bot run spelling docs pylint pyccel_lint coverage

**Review by sboof911 — APPROVED** [review]
Looks Good.

File: pyccel/codegen/printing/cwrappercode.py [hunk 1]
```diff
@@ -240,6 +240,8 @@ def _print_PyArgKeywords(self, expr):
     def _print_PyModule_AddObject(self, expr):
         name = self._print(expr.name)
         var  = self._print(expr.variable)
+        if not expr.variable.dtype is PyccelPyObject():
```
Comment (yguclu) [inline pyccel/codegen/printing/cwrappercode.py:243]: More Pythonic is writing
```suggestion
        if expr.variable.dtype is not PyccelPyObject():
```

File: tests/epyccel/test_epyccel_classes.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+from pyccel.epyccel import epyccel
+
+def test_class_import(language):
+    class A:
+        def __init__(self : 'A'):
+            pass
+
+    epyc_A = epyccel(A)
+
+    assert isinstance(epyc_A, type)
```
Comment (yguclu) [inline tests/epyccel/test_epyccel_classes.py:11]: Shouldn't `epyc_A` be of type `A` just like in Python? If not, code that checks `isinstance(obj, A)` will fail to work after pyccelization.

**Review by yguclu — CHANGES_REQUESTED** [review]
Looks good to me

File: pyccel/codegen/printing/cwrappercode.py (hunk 1 above)
Comment (EmilyBourne) [reply pyccel/codegen/printing/cwrappercode.py]: Huh, pylint used to raise an error about this

File: tests/epyccel/test_epyccel_classes.py (hunk 2 above)
Comment (EmilyBourne) [reply tests/epyccel/test_epyccel_classes.py:11]: No. We are returning the Pyccelised class A not an instance of that class so the result must be a type. If we want the class methods to call Pyccelised functions I don't think the resulting class can be equivalent to A

Comment (yguclu) [reply tests/epyccel/test_epyccel_classes.py:11]: Ah sorry, this is not an instance of A but A itself!

**Review by yguclu — APPROVED** [review]

**Pull request merged**